### PR TITLE
Plot linear decision boundaries

### PR DESCRIPTION
**Pull request by jbfiot**
### Code changes
* Plots the linear decision boundaries on top of low dim embeddings (all credits to @Atcold: see [here](https://github.com/Atcold/NYU-DLSP21/pull/32#issuecomment-1149411240) and [there](https://github.com/Atcold/NYU-DLSP21/pull/32#issuecomment-1149421739))
* Minor code cleaning

### Test
Local test in jupyter lab:
image

### CODE REVIEW

**Review by Atcold — CHANGES_REQUESTED** [review]
Can you please undo the replacement of single quotes with double quotes?
Also, leave those inline if-statements alone, plz 😛
Finally, use the context switch instead of `detach` below:

```python
def get_layer_outputs(name):
    def hook(model, input, output):
        layer_outputs[name] = output.detach()
```

**Comment by jbfiot** [comment]
> Can you please undo the replacement of single quotes with double quotes? Also, leave those inline if-statements alone, plz 

Done ^^  


> Finally, use the context switch instead of `detach` 

I removed `detach` in the hook and added the context switch for the model call (please see [f2aabfa](https://github.com/Atcold/NYU-DLSP21/pull/34/commits/f2aabfa60492bb980eb719c54fca5ea9715a2e3a)). Tested again the notebook, still working :)

**Review by Atcold — APPROVED** [review]

**Comment by Atcold** [comment]
Great! Thanks so much!
